openedx/openedx-platform
# feat: add API endpoint for asset usage search

## Description

This PR adds an API endpoint that fetches where assets are used within a specific course. The API is used by the course authoring MFE's new asset page. This change impacts Course Authors.

## Supporting information

JIRA Ticket: [TNL-10962](https://2u-internal.atlassian.net/browse/TNL-10962)

## Testing instructions

1. Add an asset to a course block
2. Navigate to the course- authoring asset page
3. Find asset and open info modal
4. Check that usage has correct subsection, unit, and block location for asset
5. Open an asset not used in the course
6. Check that it has a not used message.

Specific video block test
1. Add a handout to a video
2. Navigate to the course- authoring asset page
3. Find asset  that was added as a handout and open info modal
4. Check that usage has correct subsection, unit, and block location for asset

## Deadline

None

## Other information

course-authoring [PR #573](https://github.com/openedx/frontend-app-course-authoring/pull/573) is dependent on this change
